Makefile

```diff
@@ -100,6 +100,21 @@ ifeq "$(WITH_SS11)" "1"
 endif
 
 
+# Separate out NGC vs ROCM base images only because it could impact which
+# tools we add, such as nccl vs rccl, in the HPC Dockerfile. Note we could
+# likely modify this to work for both and have a separate flag to specify
+# nccl/rccl, etc, to keep things cleaner.
+ifneq ($(USER_NGC_BASE_IMAGE),)
+        $(shell echo "$(USER_NGC_BASE_IMAGE)" > tmp-user.txt)
+        USER_NGC_IMAGE_REPO=$(shell echo "$(USER_NGC_BASE_IMAGE)" | awk 'BEGIN{FS=OFS="/"}{NF--; print}')
+        USER_NGC_IMAGE_NAME=$(shell echo "$(USER_NGC_BASE_IMAGE)" | awk -F "/" '{print $$NF}' | awk -F ":" '{print $$1}')
+        USER_NGC_IMAGE_VER=$(shell echo "$(USER_NGC_BASE_IMAGE)" | awk -F "/" '{print $$NF}' | awk -F ":" '{print $$NF}')
+        USER_NGC_IMAGE_HPC=$(USER_NGC_IMAGE_REPO)/$(USER_NGC_IMAGE_NAME)-hpc:$(USER_NGC_IMAGE_VER)
+        USER_NGC_IMAGE_SS=$(USER_NGC_IMAGE_REPO)/$(USER_NGC_IMAGE_NAME)-hpc-ss:$(USER_NGC_IMAGE_VER)
+        USER_NGC_IMAGE_SIF=$(shell echo "$(USER_NGC_BASE_IMAGE)" | sed s,'/','-',g | sed s,':','-',g)
+endif
```

Comment:
Some interesting uses of `awk` above, you trying to show off?  :-) 

For last line, you could combine the two `sed` invocations to:
```USER_NGC_IMAGE_SIF=$(shell echo "$(USER_NGC_BASE_IMAGE)" | sed 's,[/:],-,g'```